projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DeploymentService.java

```diff
@@ -45,10 +45,20 @@ public class DeploymentService {
   private final DataJobMetrics dataJobMetrics;
   private final SupportedPythonVersions supportedPythonVersions;
 
-  public Optional<JobDeploymentStatus> readDeployment(String jobName) {
+  public Optional<JobDeploymentStatus> readOptionalDeployment(String jobName) {
     return jobImageDeployer.readScheduledJob(jobName);
   }
 
+  public JobDeploymentStatus readDeployment(String jobName) {
+    return jobImageDeployer
+        .readScheduledJob(jobName)
+        .orElseThrow(() -> new DataJobDeploymentNotFoundException(jobName));
```

Comment:
nit: may reuse `readDeploymentOptional`; so that in case the implementation changes (readScheduledJob), you only need to edit once method; and also prevent duplicate code